### ISSUE TITLE
Backup fails verification

### ISSUE THREAD

**Issue by rperezlo** [skleeschulte/basic-to-passport-auth-http-proxy#17]
Running docker on DSM 7 on a ds720+. 10 GB of RAM (I point this because I wonder if having more than 8 GB could cause some data corruption).

For a while It works as expected, but, after some weeks of daily backups, HyperBackup warns that the backup is corrupted and that it's only available for restore. The log shows some hundreds files corrupted. If I manually check the corrupted files, I can't open them. If I check other non corrupted files, they open fine.

The HB verification is not continuous, so it's possible that those files are corrupted from day one.

**Comment by poldixd** [comment]
Same here. My backup fails since 21. November. I can‘t create a backup on a DS216+II DSM7.

**Comment by skleeschulte** [comment]
Please give version [v0.2.0-beta.0](https://github.com/skleeschulte/basic-to-passport-auth-http-proxy/tree/v0.2.0-beta.0) a try. Updated all dependencies and updated the Docker image to be based on `node:12` instead of `node:10-alpine`.

**Comment by skleeschulte** [comment]
Although, what you describe might as well be the result of larger files not being uploaded correctly due to limitations of OneDrive and the Synology client, see #11 .

**Comment by rperezlo** [comment]
Thanks for your answers. Files failing the verification are of all sizes, from 50 bytes to 95 MB. There are 721 files corrupted out of roughly 500K, but there may be more.

I'll try the new version and let you know if there are any changes, but it can take weeks before it fails again.

**Comment by rperezlo** [comment]
**Edit**: Forget it, I was changing the external port instead of the local port. Besides that, I've decided to stop the 1.4 container and I'm running only the beta 2 now.

The backup is running now, I'll let you know the outcome and if there are any more issues.


**Old message:** (((Noob question. How can I run two versions of the container at the same time? I have another backup job running with version 1.4 and I don't want to stop it (yet).

I've tried to change the port in the beta container and HB to 3001 but that doesn't seem to be enough, it returns a 'can't connect to the target' error.)))

**Comment by skleeschulte** [comment]
Alright. If I understand #11 correctly, the backup files themselves are all capped to 50 MB, while some databse-files of the backup grow with the size of the backup and time. This might be an explanation why at some point arbitrary files fail to be backed up, as the backup's database file cannot be updated anymore. This is a wild guess though. I switched to a QNAP NAS some time ago - it has other flaws, but the backup to OneDrive works without problems with QNAPS HBS software.

I tested the latest version of the proxy, v0.2.0-beta.0, with WinSCP, and I get the 413 Request Entity too large error there too when uploading large files. As far as I know, this cannot be easily fixed when using WebDAV, as [WebDAV does not allow chunked uploading](https://stackoverflow.com/questions/20969331/standard-method-for-http-partial-upload-resume-upload) of files. The go-to solution would be to use the [Microsoft Graph API](https://docs.microsoft.com/en-us/onedrive/developer/rest-api/?view=odsp-graph-online) instead of WebDAV to access OneDrive, which allows to [upload large files in an upload session](https://docs.microsoft.com/en-us/onedrive/developer/rest-api/api/driveitem_createuploadsession?view=odsp-graph-online). This is what [rclone OneDrive](https://rclone.org/onedrive/) does.

**Comment by rperezlo** [comment]
The two backups have finished correctly using the 2.0 beta. The one that failed previously is 440 GB, I have recreated it (it has taken 16 hours to complete). The other one is 800 GB and it has run it's nightly backup fine.

There seems to be no problem at all. I've checked several files and they seem to be fine. There is a 11 GB ISO file that I've just restored and it is working fine. I'll report if/when verification fails again.

BTW, both backups are encrypted.

I've checked rclone. It looks much more complicated though, and your solutions seems to work so well (until it fails verification).

Thanks for your help.
